nyc_taxi_sampling/nyc_taxi_sampling.ipynb

```diff
@@ -0,0 +1,389 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "using AWSS3\n",
+    "using FilePathsBase\n",
+    "\n",
+    "data_path = \"https://s3.amazonaws.com/nyc-tlc/trip+data/yellow_tripdata_2016-01.parquet\"\n",
+    "s3_bucket_name = get_cluster_s3_bucket_name(cluster_name)\n",
+    "\n",
+    "temp_path = download(data_path)\n",
+    "cp(\n",
+    "    Path(temp_path),\n",
+    "    S3Path(\"s3://$s3_bucket_name/nyc_tripdata.parquet\", config=Banyan.get_aws_config())\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Sample First, Compute Later\n",
+    "\n",
+    "Any computed value can be sampled with Banyan to provide a quick approximation\n",
+    "of the value by running the computation locally. Further analysis can then be done\n",
+    "locally before actually computing the value on the cluster."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# First start a session\n",
+    "session_id = start_session(\n",
+    "    cluster_name = cluster_name,\n",
+    "    nworkers = 2,\n",
+    "    sample_rate = 128,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Read the dataset\n",
+    "s3_bucket_name = get_cluster_s3_bucket_name(cluster_name)\n",
+    "df = BanyanDataFrames.read_parquet(\n",
+    "    \"s3://$s3_bucket_name/nyc_tripdata.parquet\",\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Filter all trips with distance longer than 1.0, group by passenger count,\n",
+    "# and get the average trip distance for each group.\n",
+    "long_trips = filter(\n",
+    "    row -> row.trip_distance > 1.0,\n",
+    "    df\n",
+    ")\n",
+    "gdf = groupby(long_trips, :passenger_count)\n",
+    "trip_means = combine(gdf, :trip_distance => mean)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Take a sample of the result to get a quick approximatation. This computation\n",
+    "# runs locally on your computer\n",
+    "trip_means_sample = sample(trip_means)\n",
+    "println(trip_means_sample)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Now, compute the actual result using the whole dataset on the cluster.\n",
+    "trip_means_whole = compute(trip_means)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# End the session\n",
+    "end_session()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Compute Once, Sample Later\n"
```

Comment:
"Sample and Compute Locally"

Banyan makes it easy to sample any computation and continue computation locally on your laptop after shutting down cloud resources. This reduces your cloud costs and data center emissions at the same time.